src/libcore/iter/iterator.rs

```diff
@@ -1426,12 +1426,7 @@ pub trait Iterator {
         Self: Sized,
         F: FnMut(Self::Item) -> bool
     {
-        for x in self {
-            if f(x) {
-                return true;
-            }
-        }
-        false
+        !self.all(move |x| !f(x))
```

Comment:
This is a breaking change in semantics for code like this:

```
impl Iterator for _ {
    fn all(...) -> bool where ... { !self.any(move |x| !f(x)) }
    // no any def’n.
}
```

Currently it would just work, but after this change it would become an infinite loop.

I noticed this, because this is a common issue in Haskell, which necessitated the “Minimal complete definition” documentation sections.